packages/core/src/storage.ts

```diff
@@ -85,30 +71,48 @@ export interface StorageAPI {
    * storage.store(readable, { headers: encoder.headers })
    *   .then((request) => console.log(request));
    */
-  store: (data: File | Readable, config?: OpenOptions) => Promise<Request>;
+  store: (
+    data: File | Readable,
+    config?: OpenOptions
+  ) => Promise<StorageRequest>;
 
   /**
    * Retrieve the status of a storage request from a remote provider.
    *
    * @param id The id of the storage request, as returned from the `store` function.
-   * @returns Promise that resolves to a storage RequestInfo object.
+   * @returns Promise that resolves to a StatusRequest object.
    */
-  status: (id: string) => Promise<RequestInfo>;
+  status: (id: string) => Promise<StatusRequest>;
+
+  /**
+   * Retrieve the status of a storage request from a remote provider.
+   *
+   * @param cid The cid of the target data, as returned from the `store` function.
+   * @returns Promise that resolves to an array of storage StatusRequest objects.
+   */
+  statusByCid: (id: string) => Promise<StatusRequest[]>;
```

Comment:
rather than change status, I've created a second api here that allows us to query by cid and returns an array. in case we still want to query by status of a given deal request specifically.